internal/actions/reconcile_components_test.go

```diff
@@ -0,0 +1,245 @@
+package actions
+
+import (
+	"github.com/broadinstitute/revere/internal/configuration"
+	"github.com/broadinstitute/revere/internal/statuspage"
+	"github.com/go-resty/resty/v2"
+	"github.com/jarcoal/httpmock"
+	"reflect"
+	"sort"
+	"testing"
+)
+
+func TestReconcileComponents(t *testing.T) {
+	type args struct {
+		config *configuration.Config
+		client *resty.Client
+	}
+	config := configuration.Config{
+		Verbose: false,
+		Client: struct {
+			Redirects int
+			Retries   int
+		}{Redirects: 0, Retries: 0},
+		Statuspage: struct {
+			ApiKey     string `validate:"required"`
+			PageID     string `validate:"required"`
+			ApiRoot    string
+			Components []configuration.Component
+		}{ApiKey: "key", PageID: "foo", ApiRoot: "https://localhost",
+			Components: []configuration.Component{
+				{Name: "Same", Description: "Same description"},
+				{Name: "Modified", Description: "New description"},
+				{Name: "New", Description: "A new component"},
+			},
+		},
+	}
+	client := statuspage.Client(&config)
+	tests := []struct {
+		name string
+		args args
+		// ID-component maps to use for the mock
+		start   map[string]statuspage.Component
+		end     map[string]statuspage.Component
+		wantErr bool
+	}{
+		{
+			name: "Reconciles components",
+			args: args{
+				config: &config,
+				client: client,
+			},
+			start: map[string]statuspage.Component{
+				"1": {Name: "Same", Description: "Same description", Showcase: true, Status: "operational", ID: "1", PageID: "foo"},
+				"2": {Name: "Modified", Description: "Old description", Showcase: true, Status: "operational", ID: "2", PageID: "foo"},
+				"3": {Name: "Deleted", Description: "To be deleted", Showcase: true, Status: "operational", ID: "3", PageID: "foo"},
+			},
+			end: map[string]statuspage.Component{
+				"1": {Name: "Same", Description: "Same description", Showcase: true, Status: "operational", ID: "1", PageID: "foo"},
+				"2": {Name: "Modified", Description: "New description", Showcase: true, Status: "operational", ID: "2", PageID: "foo"},
+				"4": {Name: "New", Description: "A new component", Showcase: true, Status: "operational", ID: "4", PageID: "foo"},
+			},
+		},
+	}
+	for _, tt := range tests {
```

Comment:
slight preference for a name like `testCase` or even `case` here but not a big deal

Reply:
I actually feel the same way, but this part is codegen from GoLand. I can't figure out a way to change that generation to use a nicer variable name, and it seems like it is doing "intelligent" things to avoid variable collisions. Maybe once the code is more stable a global find+replace could clean it up, I just figure maybe we stay consistent until then.